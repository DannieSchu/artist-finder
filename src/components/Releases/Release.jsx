import React from 'react';
import PropTypes from 'prop-types';
import styles from '../Artists/Artist.jsx';

const Release = ({ title, date, url = '/music-icon.svg' }) => {
  return (
    <table className={styles.Artist}>
      <tr>
        <th>Release</th>
        <td>{title}</td>
      </tr>
      <tr>
        {date && <td>{date}</td>}
        {date && <th>Date</th>}
      </tr>
      <image src={url} alt={url ? name : 'Default image'} />
    </table>
  );
};

Release.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.string,
  url: PropTypes.string
};

export default Release;
