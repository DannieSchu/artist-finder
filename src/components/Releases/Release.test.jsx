import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release.jsx';

describe('Release', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Release
        title="Wonderful Town"
        date="1953" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
