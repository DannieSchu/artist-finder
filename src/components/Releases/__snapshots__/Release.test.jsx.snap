// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release matches a snapshot 1`] = `
<table>
  <tr>
    <th>
      Release
    </th>
    <td>
      Wonderful Town
    </td>
  </tr>
  <tr>
    <td>
      1953
    </td>
    <th>
      Date
    </th>
  </tr>
  <image
    alt=""
    src="/music-icon.svg"
  />
</table>
`;
